src/EditorFeatures/TestUtilities2/Intellisense/IntelliSenseTestState.vb

```diff
@@ -0,0 +1,15 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.ComponentModel.Composition
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
+    <Export(GetType(IIntelliSenseTestState))>
+    <PartNotDiscoverable>
```

Comment:
What is going on here? I don't think you're doing something _wrong_, I just suspect a comment will go along way to help the next person maintaining this. (Your commit message describing this commit is also empty beyond the title which isn't helping either.)

Reply:
When `[PartNotDiscoverable]` is applied, the MEF container will not contain the part if you simply add an assembly to the test container. It will only be included if you explicitly add it, e.g. by calling `.WithPart(typeof(IntelliSenseTestState))`. Once you explicitly include the type, the rest of the MEF attributes applied to it are used to determine the export contracts and/or metadata (the same as any other exported item).

In general, `PartNotDiscoverable` should be applied to all MEF parts created for test purposes. Otherwise, you run the risk of the part being discovered and used by an unrelated test, potentially negatively impacting the test results. In the past, constructor injection was used for several such test cases so the `[Export]` attribute was never applied to these test parts. However, now that test parts are placed directly in the MEF container we need to apply the new attribute to limit the tests where it's used.

Reply:
A good rule of thumb to consider as you get more familiar with this is the following:

> If you see a `[Export]` applied to a type that doesn't appear to be a type we want to use by default in production, it may be missing a `PartNotDiscoverable` attribute.